src/corona_hakab_model/manager.py

```diff
@@ -152,6 +149,9 @@ def step(self):
         medical_machine_step_result = self.medical_state_manager.step(new_infection_cases.keys())
         self.new_sick_counter = medical_machine_step_result['new_sick']
 
+        for agent in self.agents:
+            agent.update_contagiousness()
```

Comment:
Too wastefull... you need to update only if the agent is infectiouss

Reply:
fixed (for each agent, checked if the agent's state is instance of contagious state)

Reply:
Or just go oever the ones that are not susptible :D